src/Npgsql/NpgsqlParameterCollection.cs

```diff
@@ -568,11 +567,11 @@ internal bool HasOutputParameters
             }
         }
 
-        static NpgsqlParameter Cast(object value)
+        static NpgsqlParameter Cast(object? value)
```

Comment:
This is converting a nullable parameter to a non-nullable return value... not good. It should accept and return the same nullability, so probably do the check before calling this method.

Reply:
Check how it's used. This method produces a non-nullable value which later passed to strongly typed methods accepting non-nullable parameters.

Reply:
I don't really understand... All invocations pass non-nullable `object` except for AddRange - which throws if Cast returns null. Why not just have Cast() accept and return non-nullable, and do the null check in AddRange before calling Cast?

Reply:
Null check are performed by methods after check. For example `Add(object)` calls `Cast` and then `Add(NpgsqlParameter)`. So there are no double null checks.

Reply:
OK, I don't care enough - we need to merge this PR ASAP.